### ISSUE TITLE
Install.ps1 hangs forever while trying to connect through adb

### ISSUE THREAD

**Issue by Denniso2** [LSPosed/MagiskOnWSA#101]
Got a bit of a weird issue going on.Installation proceeds fine but hangs forever while waiting for an adb connection.Amazon appstore is open and developer mode is turned on.
This is the powershell output:
![image](https://user-images.githubusercontent.com/42325785/142850932-bb8ba5ba-9b77-46c2-a444-0523767f4090.png)
And this the subsystem for android settings:
![image](https://user-images.githubusercontent.com/42325785/142851081-794642e1-2772-47d5-97ac-ff94583d7d30.png)
If I restart and try to connect manually ```cannot connect to 127.0.0.1:58526: No connection could be made because the target machine actively refused it. (10061)```
If I try to connect after enabling wireless debugging through the vm ip:port I get ```failed to connect to 172.23.231.76:35005```

**Comment by yujincheng08** [comment]
Could you try to tick the dev mode off and on again and see if adb connects afterward?

**Comment by Denniso2** [comment]
> Could you try to tick the dev mode off and on again and see if adb connects afterward?

I have already done that and still no change.I have reinstalled wsa a couple of times and no change.I tried disabling windows firewall but seems unrelated.If there are any logs I should provide please let me know.

**Comment by avertv** [comment]
and if so ? `adb connect 172.23.231.76:5555 `

**Comment by Denniso2** [comment]
> and if so ? `adb connect 172.23.231.76:5555 `
I can successfully connect using the port 5555.Any ideas why I can connect with localhost?

**Comment by yujincheng08** [comment]
58526 is a mapped port to 5555 of the VM. A possible reason is that other applications already use 58526, and WSA cannot bind to it anymore. It's a common issue somehow because Hyper-V sometimes uses random ports for NAT. Or, maybe WSA is blocked by the firewall.

**Comment by onomatopellan** [comment]
I had the same problem and indeed the problem was the excluded ports reserved for Hyper-v.
Running `netsh int ipv4 show excludedportrange protocol=tcp` you can see port 58526 is in the exclusion range 58475-58574  so it can't be used.
```
Protocol tcp Port Exclusion Ranges

Start Port    End Port
----------    --------
      2869        2869
      5357        5357
     10243       10243
     49706       49805
     49806       49905
     50000       50059     *
     50428       50527
     58475       58574

* - Administered port exclusions.
```
